_posts/2021-10-01-debezium-server-iceberg.adoc

```diff
@@ -0,0 +1,163 @@
+---
+layout: post
+title:  Using Debezium to Create ACID Data Lake
+date:   2021-10-01
+tags: [ debezium, iceberg, datalake, lakehouse ]
+author: isimsek
+---
+
+Do you need to build Lakehouse with near realtime data pipeline, do you want it to support ACID transactions and updates on data lake?
+It is possible with "Debezium Server Iceberg" without any dependency to Apache Kafka or Apache Spark applications.
+
++++<!-- more -->+++
+
+==== Debezium
+
+https://debezium.io/[Debezium] is an open source distributed platform for change data capture.
+Debezium extracts realtime database changes as json, avro, protobuf events and delivers to event streaming platforms
+(Kafka, Kinesis, Google Pub/Sub, Pulsar are just some of https://debezium.io/documentation/reference/operations/debezium-server.html#_sink_configuration[supported sinks],
+it provides simple interface to https://debezium.io/documentation/reference/operations/debezium-server.html#_implementation_of_a_new_sink[implement new sink].
+
+==== Apache Iceberg
+
+https://iceberg.apache.org/[Apache Iceberg] is an open table format for huge analytic datasets.
+Iceberg adds tables to compute engines including Spark, Trino, PrestoDB, Flink and Hive using a high-performance table format that works just like a SQL table.
+It supports ACID Inserts, Row level Deletes/Updates. It has flexible foundation and comes with https://iceberg.apache.org[many features].
+
+== Debezium Server Iceberg
+
+**Debezium Server Iceberg** project implements new consumer which uses Iceberg Java API to consume events.
+Iceberg consumer converts received events to iceberg format and commits them to destination iceberg table. With the consumer it's possible to use any cloud store/any catalog supported by iceberg.
+
+on high level data processing works like below ::
+for each batch of events
+* it groups events per event destination,
+* for each destination, events are converted to iceberg records. at this step event schema used to map data to iceberg record (`debezium.format.value.schemas.enable` should be enabled(true)).
+* After events converted to iceberg records, they are saved to iceberg parquet files(iceberg data and delete files(for upsert)).
+* as a last step these files are committed to destination table(uploaded to destination storage)) using iceberg java API.
+
+If destination table not found in the destination catalog consumer will try to create it using event schema and key schema(record key/PK).
+
+image::../assets/images/debezium-iceberg.png[Debezium-Iceberg]
+
+Currently, Iceberg Consumer supports only json events. With json events event schema must be enabled to do correct type conversion to iceberg record.
+
+example configuration::
```

Comment:
"Here is a complete example configuration for using Debezium Server with the Iceberg adaptor:"